seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/AbstractJdbcCatalog.java

```diff
@@ -277,15 +285,24 @@ public List<String> listDatabases() throws CatalogException {
 
     @Override
     public boolean databaseExists(String databaseName) throws CatalogException {
-        checkArgument(StringUtils.isNotBlank(databaseName));
-
-        return listDatabases().contains(databaseName);
+        if (StringUtils.isBlank(databaseName)) {
+            return false;
+        }
+        if (!SYS_DATABASES.isEmpty() && SYS_DATABASES.contains(databaseName)) {
+            return false;
+        }
+        return queryExists(
+                getUrlFromDatabaseName(databaseName), getDatabaseWithConditionSql(databaseName));
```

Comment:
> Do we need to keep the rollback strategy? If Catalog does not support the `getDatabaseWithConditionSql` method, we can continue to use the old check way.

I think it's no need to use this strategy because  all jdbc catalogs are supported this way.

Reply:
But if the user adds a custom Catalog, it needs to be re-adapted. This is my concern.

Reply:
> But if the user adds a custom Catalog, it needs to be re-adapted. This is my concern.

I think the old way is a wrong way to use, should be abandoned, if the user needs to customize the catalog should be a need to consider this issue.

Reply:
But in fact, the old way only has performance issues. If the user does not have a lot of tables, there is no problem. This PR solves the performance problem but should not invalidate the functions of the custom Catalog (It can not worked after merge this PR if do not change the code).

Reply:
> But in fact, the old way only has performance issues. If the user does not have a lot of tables, there is no problem. This PR solves the performance problem but should not invalidate the functions of the custom Catalog (It can not worked after merge this PR if do not change the code).

In fact, even the original method is also executed the SQL first to get all the table names, and if the SQL  is not supported, the original method is also not supported.

Reply:
The old Catalog doesn't implement `getDatabaseWithConditionSql` method. So when invoke this method, the `UnsupportedOperationException` would be throwed.

The difference is that the user need or no need to modify the code

Reply:
> The old Catalog doesn't implement `getDatabaseWithConditionSql` method. So when invoke this method, the `UnsupportedOperationException` would be throwed.
> 
> The difference is that the user need or no need to modify the code

Okay, you convinced me.